components/micro-gateway-core/src/main/ballerina/gateway/constants/api_security_constants.bal

```diff
@@ -141,14 +141,15 @@ public function getAuthenticationFailureMessage(int errorCode) returns string {
     return errorMessage;
 }
 
-public function getFailureMessageDetailDescription(int errorCode, string errorMessage) returns string {
+public function getFailureMessageDetailDescription(http:FilterContext context, int errorCode, string errorMessage) returns string {
     string errorDescription = errorMessage;
     if (API_AUTH_INCORRECT_API_RESOURCE == errorCode) {
         errorDescription += DESCRIPTION_SEPARATOR + API_AUTH_INCORRECT_API_RESOURCE_DESCRIPTION;
     } else if (API_AUTH_INCORRECT_API_RESOURCE == errorCode) {
         errorDescription += DESCRIPTION_SEPARATOR + API_AUTH_ACCESS_TOKEN_INACTIVE_DESCRIPTION;
     } else if (API_AUTH_MISSING_CREDENTIALS == errorCode) {
-        errorDescription += DESCRIPTION_SEPARATOR + API_AUTH_MISSING_CREDENTIALS_DESCRIPTION;
+        string authHeaderName = getAuthorizationHeader(serviceAnnotationMap[getServiceName(context.serviceName)] ?: []);
+        errorDescription += DESCRIPTION_SEPARATOR + "Make sure your API invocation call has a header: \"" + authHeaderName + "\"";
```

Comment:
Avoid string initialization for a single character
```suggestion
        errorDescription += DESCRIPTION_SEPARATOR + "Make sure your API invocation call has a header: \"" + authHeaderName + '"';
```

Reply:
ince this is ballerina lets strict with double quotation